### PR TITLE
[stable-3_2_1] pkp#pkp-lib#5565 Submission preview feature implementation

### PR DESCRIPTION
**Pull request by Vitaliy-1**
Allows submission preview starting from a copyediting stage according to the initial implementation by @majaroenne. Note that this is for the stable-3_2_1 branch.

### CODE REVIEW

File: pages/workflow/WorkflowHandler.inc.php [hunk 1]
```diff
@@ -119,7 +119,7 @@ function setupIndex($request) {
 		$workflowData['assignToIssueUrl'] = $assignToIssueUrl;
 		$workflowData['issueApiUrl'] = $issueApiUrl;
 		$workflowData['sectionWordLimits'] = $sectionWordLimits;
-		$workflowData['i18n']['schedulePublication'] = __('editor.submission.schedulePublication');
```
Comment (Vitaliy-1) [inline pages/workflow/WorkflowHandler.inc.php:122]: This localization is already added in pkp-lib that is used in 3.2.1 branch

File: locale/en_US/submission.po [hunk 2]
```diff
@@ -158,6 +158,9 @@ msgstr "The requested citation format could not be retrieved."
 msgid "submission.metadataDescription"
 msgstr "These specifications are based on the Dublin Core metadata set, an international standard used to describe journal content."
 
+msgid "submission.viewingPreview"
+msgstr "This is a preview and has not been published. <a href=\"{$url}\">View submission</a>"
```
Comment (NateWr) [inline locale/en_US/submission.po:162]: :+1:

File: templates/frontend/objects/article_details.tpl [hunk 3]
```diff
@@ -69,8 +69,15 @@
 {/if}
 <article class="obj_article_details">
 
+	{* Indicate if this is only a preview *}
+	{if $article->getData('status') !== $smarty.const.STATUS_PUBLISHED || $publication->getData('status') !== $smarty.const.STATUS_PUBLISHED}
```
Comment (NateWr) [inline templates/frontend/objects/article_details.tpl:73]: Only check the publication here. In theory, a preview should work for a second version, where the article is published but the publication is not.

**Review by NateWr — CHANGES_REQUESTED** [review]
Thanks @Vitaliy-1! Looks good. Just one minor change.  :+1:

File: templates/frontend/objects/article_details.tpl (hunk 3 above)
Comment (Vitaliy-1) [reply templates/frontend/objects/article_details.tpl:73]: Good point, I haven't thought about several publications! Preview button appears only if the submission isn't published and is at least on copyediting stage:
```
<pkp-button v-else-if="submission.status !== getConstant('STATUS_PUBLISHED') && submission.stageId >= getConstant('WORKFLOW_STAGE_ID_EDITING')"
```
Does it make sense to make an additional check here, e.g., if the current publication isn't published, show a preview button that links to that publication (not just to the submission page)?

Comment (NateWr) [reply templates/frontend/objects/article_details.tpl:73]: Yes, that'd be great if it's not too much trouble.

File: templates/workflow/workflow.tpl [hunk 4]
```diff
@@ -50,12 +50,21 @@
...
+				<pkp-button v-else-if="submission.status !== getConstant('STATUS_PUBLISHED') && submission.stageId >= getConstant('WORKFLOW_STAGE_ID_EDITING')"
+					element="a"
+					:label="i18n.preview"
+					:href="submission.urlPublished"
+				></pkp-button>
+				<pkp-button v-if="submission.status === getConstant('STATUS_PUBLISHED') && workingPublication.status !== getConstant('STATUS_PUBLISHED')"
+					element="a"
+					:label="i18n.preview"
+					:href="workingPublication.urlPublished"
+				></pkp-button>
```
Comment (NateWr) [inline templates/workflow/workflow.tpl:67]: I think for this preview to make sense for a separate version, we need to put the buttons down in the publication tab, next to where we Publish / Schedule for Publication. Can you make that change here and in the `master` PR?

Comment (Vitaliy-1) [reply templates/workflow/workflow.tpl]: Done!

**Pull request merged**